src/main/java/io/jenkins/plugins/ml/IPythonUserConfig.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 Loghi Perinpanayagam.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml;
+
+public class IPythonUserConfig {
+    private String serverGatewayAddress;
+    private long iPythonLaunchTimeout;
+    private long maxResult;
+
+    /**
+     * Constructor for default configuration
+     */
+    IPythonUserConfig() {
+        this.serverGatewayAddress = "127.0.0.1";
+        this.iPythonLaunchTimeout = 1000;
+        this.maxResult = 3;
+    }
```

Comment:
Is this used only for testing? If so we can a) remove the default constructor and b) make the class fields `final` removing the setters. This limits risk with concurrency later, or other bugs in case we accidentally re-use the user config object.

Reply:
I used the default constructer which does not need to specify the configuration. If we do a) and b), we can initialize the default settings in the `Builder` by checking whether the parameters are empty or not.